src/dpms/dpmsWidget.tsx

```diff
@@ -612,6 +631,11 @@ fetching database name from fully qualified name structure */
   useEffect(() => {
     getDatabaseDetails();
   }, [dataprocMetastoreServices]);
+  useEffect(() => {
+    databaseNames.map(async (db: string) => {
```

Comment:
don't do this, you are serially waiting on an array of promises.  use something like promise.all instead.

Also don't use useEffect to handle callbacks, just make these calls in the callback (the thing that calls setDatabaseNames)

Reply:
resolved